### ISSUE TITLE
Delete dependents first

### ISSUE THREAD

**Issue by sabrehagen** [rebuy-de/aws-nuke#38]
Some AWS resources cannot be deleted before their dependent services are deleted. Is it possible for you to add the ability to remove resources in a specific order to ensure dependents are removed before their dependencies?

```
us-east-1 - IamRole - 'storage-v2-jdabilio-IamRoleLambda-129UJS1B57Z30' - DeleteConflict: Cannot delete entity, must delete policies first.
	status code: 409, request id: 6e327e18-bd22-11e7-995f-afa65320bd29
us-east-1 - IamRole - 'storage-v2-jdabilio-IamRoleLambda-137J19IBXLR69' - DeleteConflict: Cannot delete entity, must delete policies first.
	status code: 409, request id: 6e5fa903-bd22-11e7-995f-afa65320bd29
us-east-1 - IamRole - 'storage-v2-jdabilio-ScalingRole-N7NQ5CCTG0TK' - DeleteConflict: Cannot delete entity, must delete policies first.
	status code: 409, request id: 6e8d482f-bd22-11e7-995f-afa65320bd29
us-east-1 - IamRole - 'storage-v2-strgdel-IamRoleLambda-397JQ1MQUGNX' - DeleteConflict: Cannot delete entity, must delete policies first.
```

I know of [dependency-graph](https://github.com/jriecken/dependency-graph) in the `npm` ecosystem, but not one for Go.

**Comment by svenwltr** [comment]
The idea is not to care about the dependencies at all. The dependency graph would be very complex. Furthermore, there might be dependencies within the same resource type (eg. Security Groups depending on other Security Groups). Resolving these dependencies would be a never-ending task.

We solve this problem by just retrying until everything is gone. With your example: It fails to delete the IAM Role in the first step, because it depends on an IAM Policy. *aws-nuke* doesn't stop there, but continues with trying to delete all other resources. Eventually it will delete the dependent IAM Policy and will be able to delete the IAM Role in the next iteration.

*TL;DR* - Ignore the errors and wait for aws-nuke to finish. If there are still leftovers the probably come from missing resource types.
